wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/Views/E2EIdentityCertificateDetailsView.swift

```diff
@@ -0,0 +1,152 @@
+            Spacer()
+            SwiftUI.Button(
+                action: {
+                    dismiss()
+                },
+                label: {
+                    Image(.close)
+                        .foregroundColor(SemanticColors.Icon.foregroundDefaultBlack.swiftUIColor)
+                }
+            ).padding()
+        }.background(SemanticColors.View.backgroundDefault.swiftUIColor)
+    }
+
+    private var certificateView: some View {
+        ScrollView {
+            Text(certificateDetails)
+                .font(FontSpec.smallFont.swiftUIFont.monospaced())
+                .padding()
+                .frame(maxHeight: .infinity)
+        }
+    }
+
+    private var downloadImageButton: some View {
+        SwiftUI.Button(
+            action: {
+                performDownload?()
+            },
+            label: {
+                Image(.download)
+                    .foregroundColor(SemanticColors.Icon.foregroundDefaultBlack.swiftUIColor)
+            }
+        )
+    }
+
+    private var downloadButton: some View {
+        SwiftUI.Button(
+            action: {
+                performDownload?()
+            },
+            label: {
+                Text(L10n.Localizable.Content.Message.download)
+                    .font(FontSpec.normalRegularFont.swiftUIFont.bold())
+            }
+        )
+        .foregroundColor(SemanticColors.Label.textDefault.swiftUIColor)
+    }
+
+    private var moreButton: some View {
+        SwiftUI.Button(
+            action: {
+                isMenuPresented.toggle()
+            },
+            label: {
+                Image(.more)
+                    .foregroundColor(SemanticColors.Icon.foregroundDefaultBlack.swiftUIColor)
+                    .padding(.trailing, ViewConstants.Padding.standard)
+            }
+        )
+    }
+
+    private var copyToClipboardButton: some View {
+        SwiftUI.Button(
+            action: {
+                performCopy?(
+                    certificateDetails
+                )
+            },
+            label: {
+                Text(L10n.Localizable.Device.Details.CertificateDetails.copyToClipboard)
+                    .foregroundColor(SemanticColors.Label.textDefault.swiftUIColor)
+            }
+        )
+        .foregroundColor(SemanticColors.Label.textDefault.swiftUIColor)
+    }
+
+    var body: some View {
+        titleView
+        certificateView
+        .safeAreaInset(edge: .bottom, spacing: .zero) {
+            VStack {
+                if isDownloadAndCopyEnabled {
+                    HStack {
+                        downloadImageButton.padding()
+                        downloadButton.padding()
+                        Spacer()
+                        moreButton
+                        .confirmationDialog("...", isPresented: $isMenuPresented) {
+                            copyToClipboardButton
+                        } message: {
+                        }
```

Comment:
supposed to be empty? can it be omitted?

Reply:
I will check if this could be omitted.